### PR TITLE
Add pick parameters subtask for the cli

### LINKED ISSUE #470: Subtask: populate-params
Arguably the hardest subtask of the plugin, since it needs to be able to parse user input into types known by ethers/solidity.

As in the v2x task, it should prompt the user to populate one parameter at a time. Note that this is poorly achieved in the v2x implementation, since things start getting shady once complex parameters like arrays appear.

### PR DESCRIPTION
**Pull request by ajsantander**
Fix #470

### CODE REVIEW

**Review by leomassazza — APPROVED** [review]

File: packages/cli/subtasks/pick-parameters.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        {
+          type: 'input',
+          name: 'userInput',
+          message: `  ${parameter.name} (${parameter.type}):`,
+        },
+      ]);
+
+      let encodedParameter;
+      try {
+        // Encode and decode to parse user input.
```
Comment (ajsantander) [inline packages/cli/subtasks/pick-parameters.js:29]: Could use a better comment here

File: packages/cli/subtasks/pick-parameters.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        encodedParameter = abiCoder.decode([parameter.type], encodedParameter);
+
+        hre.cli.functionParameters.push(encodedParameter);
+
+        parameterIndex++;
+      } catch (error) {
+        logger.warn(error);
+      }
+    }
+  }
```
Comment (mjlescano) [inline packages/cli/subtasks/pick-parameters.js:41]:
```suggestion
    for (const parameter of functionAbi.inputs) {
      const { userInput } = await inquirer.prompt([
        {
          type: 'input',
          name: 'userInput',
          message: `  ${parameter.name} (${parameter.type}):`,
        },
      ]);

      let encodedParameter;
      try {
        // Encode and decode the user's input to parse the input
        // into types acceptable by ethers.
        encodedParameter = abiCoder.encode([parameter.type], [userInput]);
        encodedParameter = abiCoder.decode([parameter.type], encodedParameter);

        hre.cli.functionParameters.push(encodedParameter);
      } catch (error) {
        logger.warn(error);
      }
    }
  }
```

**Review by mjlescano — APPROVED** [review]

File: packages/cli/subtasks/pick-parameters.js (hunk 2 above)
Comment (ajsantander) [reply packages/cli/subtasks/pick-parameters.js:41]: This does not retry parameters 😉

Comment (mjlescano) [reply packages/cli/subtasks/pick-parameters.js:41]: Ooh, gotcha :)

**Pull request merged**